packages/plugins/src/plugins/silo/siloFetcher.ts

```diff
@@ -0,0 +1,155 @@
+import { EvmNetworkIdType, PortfolioElement } from '@sonarwatch/portfolio-core';
+import BigNumber from 'bignumber.js';
+import { Cache } from '../../Cache';
+import { Fetcher, FetcherExecutor } from '../../Fetcher';
+import { getEvmClient } from '../../utils/clients';
+import {
+  platformId,
+  siloPoolsKey,
+  legacyLensAddress,
+  supplyTag,
+  borrowTag,
+  missingTokenPriceAddresses,
+} from './constants';
+import { balanceAbi } from './abis';
+import tokenPriceToAssetToken from '../../utils/misc/tokenPriceToAssetToken';
+import { SiloPool } from './types';
+import { getAmount } from '../../utils/octav/tokenFactor';
+
+function fetcher(networkId: EvmNetworkIdType): Fetcher {
+  const executor: FetcherExecutor = async (
+    owner: string,
+    cache: Cache
+  ): Promise<PortfolioElement[]> => {
+    const pools = await cache.getItem<SiloPool[]>(siloPoolsKey, {
+      prefix: platformId,
+      networkId,
+    });
+    if (!pools) return [];
+
+    const client = getEvmClient(networkId);
+
+    const poolCalls = pools.flatMap((pool) => [
```

Comment:
for better typesafety

```suggestion
    const poolCalls = pools.flatMap<
      ContractFunctionConfig<typeof balanceAbi>
    >((pool) => [
```